src/main/scala/thurloe/dataaccess/HttpSendGridDAO.scala

```diff
@@ -33,11 +34,13 @@ class HttpSendGridDAO extends SendGridDAO {
 
   def lookupPreferredEmail(userId: RawlsUserSubjectId): Future[RawlsUserEmail] =
     for {
-      contactEmail <- dataAccess.lookup(userId.value, "contactEmail")
-      email <- dataAccess.lookup(userId.value, "email")
-    } yield
-      if (contactEmail.keyValuePair.value.isEmpty) RawlsUserEmail(email.keyValuePair.value)
-      else RawlsUserEmail(contactEmail.keyValuePair.value)
+      email <- dataAccess
+        .lookup(userId.value, "contactEmail")
+        .recoverWith { _ =>
+          logger.info(s"Failed to get stored contactEmail for ${userId.value}. Defaulting to stored email for user.")
+          dataAccess.lookup(userId.value, "email")
```

Comment:
for quick debugging, I think logging the email address can be more(?) helpful since it's more human-readable. I think it is just about as unique as the userId as well. I'm not too tied to either approach though!